### ISSUE TITLE
Woking?

### ISSUE THREAD

**Issue by IsNoobgrammer** [defdet/tpu-generate#1]
Does this fork of transformers support interference on TPU ?

Model is trained similarly :->  like in your notebooks(training on tpu with spmd) with some optimizations

**Comment by defdet** [comment]
Nope, not yet. I've been trying to make inference work on TPU but have been encountering some weird errors (model generates gibberish for some reason). The best way currently is to make inference through JAX with custom build models. Available models are [llama](https://github.com/ayaka14732/llama-2-jax), [falcon](https://github.com/defdet/falcon-jax). Falcon only works for 7B though, 40 and 180B have their architecture altered. And also Mistral but without sliding window attention. I've created some notebooks that fully work: https://www.kaggle.com/code/defdet/mistral-essay-generation, https://www.kaggle.com/code/defdet/jax-llama-essay-generation, https://www.kaggle.com/code/defdet/jax-falcon-7b-generation.
Also, this is a cloned repo with some fixes to `llama_modeling.py` regarding cpu fallback with torch XLA. Hopefully it won't be needed

**Comment by defdet** [comment]
By the way, regarding your comment on Kaggle - yes, we can accumulate gradients in the training loop, it should be easy to just pogramm it in. Don't really get the point of doing so though.
About inference with torch xla SPMD - yes, that's exactly what I'm planning to complete.

**Comment by IsNoobgrammer** [comment]
> By the way, regarding your comment on Kaggle - yes, we can accumulate gradients in the training loop, it should be easy to just pogramm it in. Don't really get the point of doing so though. About inference with torch xla SPMD - yes, that's exactly what I'm planning to complete.

I tried implementing Gradient accumulation But was increasing training time by alot due to unsynced shards , 
so i was asking whether we can use it so that batch-size increases(bs*grad_steps) thus allowing for 
less training time even for very large datasets (500k+)  since kaggle is giving only 20h per week
